src/test/java/jenkins/plugins/logstash/LogstashOutputStreamTest.java

```diff
@@ -118,4 +119,17 @@ public void eolSuccessNoDao() throws Exception {
     assertEquals("Results don't match", msg, buffer.toString());
     verify(mockWriter).isConnectionBroken();
   }
+
+  @Test
+  public void writerClosedBeforeDelegate() throws Exception {
+    ByteArrayOutputStream mockBuffer = Mockito.spy(buffer);
+    new LogstashOutputStream(mockBuffer, mockWriter).close();
+
+    InOrder inOrder = Mockito.inOrder(mockBuffer, mockWriter);
+    inOrder.verify(mockWriter).close();
+    inOrder.verify(mockBuffer).close();
+
+    // Verify results
+    assertEquals("Results don't match", "", buffer.toString());
```

Comment:
That message doesn't really add anything over what the matcher would report. Can we remove it?

Reply:
Actually I think it would be better to use hamcrest here: 
```
 assertThat( buffer, hasToString("")
```